src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -180,27 +180,15 @@ protected override Conversion GetInterpolatedStringConversion(BoundExpression so
             }
             else if (collectionTypeKind == CollectionExpressionTypeKind.ImplementsIEnumerableT)
             {
-                ImmutableArray<NamedTypeSymbol> allInterfaces;
-                switch (targetType.TypeKind)
-                {
-                    case TypeKind.Class:
-                    case TypeKind.Struct:
-                        allInterfaces = targetType.AllInterfacesNoUseSiteDiagnostics;
-                        break;
-                    case TypeKind.TypeParameter:
-                        allInterfaces = ((TypeParameterSymbol)targetType).AllEffectiveInterfacesNoUseSiteDiagnostics;
-                        break;
-                    default:
-                        throw ExceptionUtilities.UnexpectedValue(targetType.TypeKind);
-                }
-
+                var allInterfaces = targetType.GetAllInterfacesOrEffectiveInterfaces();
                 var ienumerableType = this.Compilation.GetSpecialType(SpecialType.System_Collections_Generic_IEnumerable_T);
                 bool isCompatible = false;
                 foreach (var @interface in allInterfaces)
                 {
                     if (isCompatibleIEnumerableT(@interface, ienumerableType, elements, ref useSiteInfo))
                     {
                         isCompatible = true;
+                        // Continue to collect all remaining use-site information
```

Comment:
```suggestion
                        // Don't break so we collect all remaining use-site information
```
Slightly clearer what this comment means if we are a bit more specific.